### ISSUE TITLE
Set VF & AF through CALL

### ISSUE THREAD

**Issue by ronag** [CasparCG/server#776]

**Comment by 5opr4ni** [comment]
plz 2.2 ;)

**Comment by TomKaltz** [comment]
+1

**Comment by premultiply** [comment]
I am not shure if FFMPEG supports realtime filter graph changes at all.

**Comment by ronag** [comment]
@premultiply it doesn't, you need to drain it and re-create it... which is doable

**Comment by premultiply** [comment]
Doesn't it requiere new buffering, seeking etc. each time?

What is the use of it doing it in FFMPEG and not using the nice CasparCG realtime mixer?
Which filters are missing in mixer which requiere realtime adjustments?

**Comment by jesperstarkar** [comment]
> Which filters are missing in mixer which requiere realtime adjustments?

Audio filters, i.e. pan.

**Comment by 5opr4ni** [comment]
and the possibility to switch channel layout.
With @jesperstarkar and my whishes.
 `af pan=stereo|c0=c1|c1=c1`

**Comment by ronag** [comment]
> Doesn't it requiere new buffering, seeking etc. each time?

Nope.

> What is the use of it doing it in FFMPEG and not using the nice CasparCG realtime mixer?

You can do more advanced stuff in FFMPEG. And it's already implemented there.

> Which filters are missing in mixer which requiere realtime adjustments?

I think it's mostly pan.

**Comment by 5opr4ni** [comment]
and switch channel layout.

**Comment by premultiply** [comment]
Yes I know FFMPEG very well. Very powerfull and I do a lot of things with it you never thought it can be done with it. BUT it is always a pain to get it doing what you want. Constructing working filter chains will make you cry most of the time. See the problems and discussion on the colorspaces. Users may be overwhelmed...
